### PR TITLE
Remove splash screen (#1871)

### LINKED ISSUE #1871: Option to disable splash-screen
I'd really like to have an option for it.
It is annoying to see it every time for like 0,2 sec (just blinks and goes right away).
Plus it is white and makes too much contrast on my black themed system.
Currently I'm modyfying code to get rid of it.

### PR DESCRIPTION
**Pull request by aivanov-oneinc**
Fixes #1871. In this branch splash screen removed at all

What did I do to test the code and ensure quality:
 - run GitExtensions
 - make sure that splash screen does not appear

Has been tested on (remove any that don't apply):
 - GIT 2.14 and above
 - Windows 10 and above

## SynchronizationContext workaround
When I removed splash screen SynchronizationContext.Current became null. After an hour of investigation I found that it creates on first UI form creation. So I decide to create temporary empty form and get SynchronizationContext.Current. And it works.
It is some kind of workaround but there already was a workaround as commented

> // Store here SynchronizationContext.Current, because later sometimes it can be null
> //see http://stackoverflow.com/questions/11621372/synchronizationcontext-current-is-null-in-continuation-on-the-main-ui-thread

### CODE REVIEW

**Comment by RussKie** [comment]
Please have your changes rebased on top of master

**Comment by aivanov-oneinc** [comment]
Done. It was my mess with upstream syncronization.

**Comment by aivanov-oneinc** [comment]
Oh god, weird morning. I did not want to close pull request. Branch rebased on top of upstream master.

File: GitExtensions/Program.cs [hunk 1]
```diff
@@ -54,27 +54,22 @@ private static void Main()
...
             string[] args = Environment.GetCommandLineArgs();
-            FormSplash.ShowSplash();
-            //Store here SynchronizationContext.Current, because later sometimes it can be null
-            //see http://stackoverflow.com/questions/11621372/synchronizationcontext-current-is-null-in-continuation-on-the-main-ui-thread
-            GitUIExtensions.UISynchronizationContext = SynchronizationContext.Current;
-            AsyncLoader.DefaultContinuationTaskScheduler = TaskScheduler.FromCurrentSynchronizationContext();
-            Application.DoEvents();
+
+            //This form created for obtain UI synchronization context only
+            using (var empFormForSyncContext = new Form())
```
Comment (RussKie) [inline GitExtensions/Program.cs]: can be simplified to `using (new Form())`

Comment (aivanov-oneinc) [reply GitExtensions/Program.cs]: Good catch. Thanks! Done.

**Comment by aivanov-oneinc** [comment]
GitExtensionsDoc updates removed too

**Pull request merged**